### PR TITLE
refactor(layout): images & keyboard

### PR DESCRIPTION
**Pull request by artyorsh**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/kittenTricks/blob/master/CONTRIBUTING.md) guide.

 #### Short description of what this resolves:

* Refactors layout components to use asset images
* Resolves layout components keyboard overlap

### CODE REVIEW

File: src/components/ecommerce/movieDetails/movieScreenshotListItem.component.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+
+export type MovieScreenshotListItemProps = ThemedComponentProps & ImageProps;
+
+class MovieScreenshotListItemComponent extends React.Component<MovieScreenshotListItemProps> {
+
+  public render(): React.ReactNode {
```
Comment (32penkin) [inline src/components/ecommerce/movieDetails/movieScreenshotListItem.component.tsx:16]: Does it make sense to move it to the separate component?

File: src/containers/layouts/auth/signIn1/signIn1.component.tsx [hunk 2]
```diff
@@ -84,6 +79,10 @@ class SignIn1Component extends React.Component<SignIn1Props, State> {
     this.setState({ formValue });
   };
 
+  private keyboardOffset = (height: number): number => {
+    return height - 152;
```
Comment (32penkin) [inline src/containers/layouts/auth/signIn1/signIn1.component.tsx:83]: Why exactly 152?

File: src/containers/layouts/auth/signUp1/signUp1.component.tsx [hunk 3]
```diff
@@ -85,6 +81,10 @@ class SignUp1Component extends React.Component<SignUp1Props, State> {
     this.props.onEwaPress();
   };
 
+  private keyboardOffset = (height: number): number => {
+    return height - 16;
```
Comment (32penkin) [inline src/containers/layouts/auth/signUp1/signUp1.component.tsx:85]: Why not 152?

File: src/containers/layouts/auth/signUp2/signUp2.component.tsx [hunk 4]
```diff
@@ -59,6 +56,10 @@ class SignUp2Component extends React.Component<SignUp2Props, State> {
     this.props.onSignUpPress(this.state.formValue);
   };
 
+  private keyboardOffset = (height: number): number => {
+    return height - 56;
```
Comment (32penkin) [inline src/containers/layouts/auth/signUp2/signUp2.component.tsx:60]: Can it be calculated for the any of use-cases?

**Review by 32penkin — APPROVED** [review]

File: src/containers/layouts/auth/signIn1/signIn1.component.tsx (hunk 2 above)
Comment (artyorsh) [reply src/containers/layouts/auth/signIn1/signIn1.component.tsx:83]: `height` arg defines the height of the keyboard
Returning value defines total height of bottom components which can be hidden by keyboard. Currently there is no way to handle it, if not only using `onLayout` (and wrapping bottom components into one container in some cases)

File: src/components/ecommerce/movieDetails/movieScreenshotListItem.component.tsx (hunk 1 above)
Comment (artyorsh) [reply src/components/ecommerce/movieDetails/movieScreenshotListItem.component.tsx:16]: Not really. Agree with that
But it's better to do so while we have more complex `ListItem`s which are separated from `List`s

File: src/containers/layouts/auth/signUp2/signUp2.component.tsx (hunk 4 above)
Comment (artyorsh) [reply src/containers/layouts/auth/signUp2/signUp2.component.tsx:60]: > Currently there is no way to handle it, if not only using onLayout (and wrapping bottom components into one container in some cases)

**Pull request merged**